spring-cloud-config-client/src/main/java/org/springframework/cloud/config/client/ConfigServicePropertySourceLocator.java

```diff
@@ -89,6 +91,20 @@ public ConfigServicePropertySourceLocator(ConfigClientProperties defaultProperti
 	public org.springframework.core.env.PropertySource<?> locate(
 			org.springframework.core.env.Environment environment) {
 		ConfigClientProperties properties = this.defaultProperties.override(environment);
+
+		if (StringUtils.startsWithIgnoreCase(properties.getName(), "application-")) {
+			InvalidApplicationNameException exception = new InvalidApplicationNameException(
+					properties.getName());
+			if (properties.isFailFast()) {
+				throw exception;
```

Comment:
I'm not sure this is what I had in mind. logging the warning is fine. I was thinking if there was some common exception that happens we could analyze it, but not fail.

Reply:
Sorry for the delayed response on this. 

That's the thing no exception/error response is returned by config server. The server returns results that are not consistent with the expectation. So here we are adding a check in the client; If fail-fast is enabled, throw an exception which will prevent application startup, otherwise log a warning and return null.